### DISCUSSION TITLE
无法发送特定长度的字符串

### DISCUSSION THREAD

**Discussion by bridgeL** [nonebot/discussions#125] [问答]
我配置了反向ws接口的nonebot、cqhttp

但是在我向群聊发送utf-8长度（一个汉字占3位，一个字母占1位）为119、121、123、125的字符串（例如`"s"*121`）时，发送失败

私聊也有同样问题，长度为121、123、125

报错如下

> 04-10 14:29:28 [DEBUG] nonebot | OneBot V11 | Calling API send_msg
> 04-10 14:29:33 [INFO] nonebot | Matcher <Matcher from frame.lazy, type=message, priority=3, temp=False> running complete04-10 14:29:33 [ERROR] nonebot | Running matcher <Matcher from frame.lazy, type=message, priority=3, temp=False> failed.Traceback (most recent call last):
>   File "<string>", line 1, in <module>
>   File "C:\Users\myname\AppData\Local\Programs\Python\Python39\lib\multiprocessing\spawn.py", line 116, in spawn_main    
>     exitcode = _main(fd, parent_sentinel)
>   File "C:\Users\myname\AppData\Local\Programs\Python\Python39\lib\multiprocessing\spawn.py", line 129, in _main
>     return self._bootstrap(parent_sentinel)
>   File "C:\Users\myname\AppData\Local\Programs\Python\Python39\lib\multiprocessing\process.py", line 315, in _bootstrap  
>     self.run()
>   File "C:\Users\myname\AppData\Local\Programs\Python\Python39\lib\multiprocessing\process.py", line 108, in run
>     self._target(*self._args, **self._kwargs)
>   File "C:\Users\myname\AppData\Local\Programs\Python\Python39\lib\site-packages\uvicorn\subprocess.py", line 76, in subprocess_started
>     target(sockets=sockets)
>   File "C:\Users\myname\AppData\Local\Programs\Python\Python39\lib\site-packages\uvicorn\server.py", line 60, in run     
>     return asyncio.run(self.serve(sockets=sockets))
>   File "C:\Users\myname\AppData\Local\Programs\Python\Python39\lib\asyncio\runners.py", line 44, in run
>     return loop.run_until_complete(main)
>   File "C:\Users\myname\AppData\Local\Programs\Python\Python39\lib\asyncio\base_events.py", line 629, in run_until_complete
>     self.run_forever()
>   File "C:\Users\myname\AppData\Local\Programs\Python\Python39\lib\asyncio\base_events.py", line 596, in run_forever     
>     self._run_once()
>   File "C:\Users\myname\AppData\Local\Programs\Python\Python39\lib\asyncio\base_events.py", line 1890, in _run_once      
>     handle._run()
>   File "C:\Users\myname\AppData\Local\Programs\Python\Python39\lib\asyncio\events.py", line 80, in _run
>     self._context.run(self._callback, *self._args)
>   File "C:\Users\myname\AppData\Local\Programs\Python\Python39\lib\site-packages\nonebot\message.py", line 153, in _check_matcher
>     await _run_matcher(Matcher, bot, event, state, stack, dependency_cache)
>  File "C:\Users\myname\AppData\Local\Programs\Python\Python39\lib\site-packages\nonebot\message.py", line 202, in _run_matcher
>     await matcher.run(bot, event, state, stack, dependency_cache)
>   File "C:\Users\myname\AppData\Local\Programs\Python\Python39\lib\site-packages\nonebot\internal\matcher.py", line 669, 
> in run
>     await self.simple_run(bot, event, state, stack, dependency_cache)
>   File "C:\Users\myname\AppData\Local\Programs\Python\Python39\lib\site-packages\nonebot\internal\matcher.py", line 636, 
> in simple_run
>     await handler(
>   File "C:\Users\myname\AppData\Local\Programs\Python\Python39\lib\site-packages\nonebot\dependencies\__init__.py", line 
> 92, in __call__
>     return await self.call(**values)
>   File "E:\robot\PoP\src\plugins\echo.py", line 34, in test
>     await matcher.send(arg)
>   File "C:\Users\myname\AppData\Local\Programs\Python\Python39\lib\site-packages\nonebot\internal\matcher.py", line 450, 
> in send
>     return await bot.send(event=event, message=_message, **kwargs)
>   File "C:\Users\myname\AppData\Local\Programs\Python\Python39\lib\site-packages\nonebot\adapters\onebot\v11\bot.py", line 249, in send
>     return await self.__class__.send_handler(self, event, message, **kwargs)
>   File "C:\Users\myname\AppData\Local\Programs\Python\Python39\lib\site-packages\nonebot\adapters\onebot\v11\bot.py", line 177, in send
>     return await bot.send_msg(**params)
>   File "C:\Users\myname\AppData\Local\Programs\Python\Python39\lib\site-packages\nonebot\adapters\onebot\v11\bot.py", line 210, in call_api
>     return await super().call_api(api, **data)
>   File "C:\Users\myname\AppData\Local\Programs\Python\Python39\lib\site-packages\nonebot\internal\adapter\bot.py", line 118, in call_api
>     raise exception
>   File "C:\Users\myname\AppData\Local\Programs\Python\Python39\lib\site-packages\nonebot\internal\adapter\bot.py", line 95, in call_api
>     result = await self.adapter._call_api(self, api, **data)
>   File "C:\Users\myname\AppData\Local\Programs\Python\Python39\lib\site-packages\nonebot\adapters\onebot\v11\adapter.py", line 107, in _call_api
>     return _handle_api_result(
>   File "C:\Users\myname\AppData\Local\Programs\Python\Python39\lib\site-packages\nonebot\adapters\onebot\v11\utils.py", line 81, in _handle_api_result
>     raise ActionFailed(**result)
> nonebot.adapters.onebot.v11.exception.ActionFailed: <ActionFailed data=None, echo={'seq': 2}, msg=SEND_MSG_API_ERROR, retcode=100, status=failed, wording=请参考 go-cqhttp 端输出>

我的cqhttp端显示如下问题

> [WARNING]: 警告: 长消息响应延迟为 2878ms, 大于 1000ms, 可能导致无法接收/发送长消息的情况, 请检查本地网络状态.
> [WARNING]: 群消息发送失败: 账号可能被风控.

如果我发送122长度的字符串就完全正常

然而当我修改cqhttp配置文件为如下所示时，问题暂时解决

>   \# 是否强制分片发送消息
>   \# 分片发送将会带来更快的速度
>   \# 但是兼容性会有些问题
>   force-fragment: true

但是，这究竟是什么原因导致的？目前的解决方法显然是治标不治本。或许，在其他特定长度点，我又将会遇到类似的问题。。

**Comment by mnixry** [comment]
> [WARNING]: 群消息发送失败: 账号可能被风控.

字面意思，腾讯决定的，多挂几天就好了

**Reply by bridgeL** [reply]
腾讯按照字节数屏蔽的吗😂122字节就能发了。。太怪了这

**Reply by mnixry** [reply]
> 腾讯按照字节数屏蔽的吗😂122字节就能发了。。太怪了这

~~坊间有句俗话：不要试图理解风控~~

**Comment by bridgeL** [comment]
找到了相同问题，或许这是一个cqhttp的bug

[1312](https://github.com/Mrs4s/go-cqhttp/issues/1312)
